### PR TITLE
BigAnimal: Google Cloud Provider

### PR DESCRIPTION
**Pull request by drothery-edb**
## What Changed?

https://enterprisedb.atlassian.net/browse/UPM-15351

Related PRs:
GCP regions: https://github.com/EnterpriseDB/docs/pull/4195 - hasn't been reviewed but is merged
Preparing your GCP account: https://github.com/EnterpriseDB/docs/pull/4302 - reviewed and merged
Cluster networking architecture for GCP: https://github.com/EnterpriseDB/docs/pull/4320 - merged but needs review from Jason
Customizing compliance: https://github.com/EnterpriseDB/docs/pull/4327 - merged but needs review from Jason
GCP BAH: https://github.com/EnterpriseDB/docs/pull/4429 - merged - not reviewed updated regions based on UI 7/28

### CODE REVIEW

**Comment by drothery-edb** [comment]
@amrita-suresh do we want to add a note that PGD clusters aren't supported?

**Comment by amrita-suresh** [comment]
> @amrita-suresh do we want to add a note that PGD clusters aren't supported?

I would defer to @nstott who is looking into whether the PGD issues can be fixed soon

File: product_docs/docs/biganimal/release/getting_started/02_connecting_to_your_cloud/connecting_gcp.mdx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   ```
+
+1. To set up your cloud provider, run the `setup-csp` command:
+
+   ```shell
+    biganimal cloud-provider setup  
+   ```
+   The command checks for cloud account readiness and displays the results. 
+
+1. If the following readiness checks aren't met for your cloud service provider, see [Configure your Google Cloud account](/biganimal/release/getting_started/preparing_cloud_account/preparing_gcp/#configure-your-aws-account) to manually configure your cloud:
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/getting_started/02_connecting_to_your_cloud/connecting_gcp.mdx:40]: is the AWS link here intentional?

File: product_docs/docs/biganimal/release/getting_started/preparing_cloud_account/preparing_gcp/index.mdx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    | -------- | ----------- |
+    | &lt;account-id> | Google Cloud account ID of your BigAnimal deployment. |
+    | &lt;region> | Google Cloud region where your clusters are being deployed. See [Supported regions](../../overview/03a_region_support/#supported-google-cloud-regions) for a list of possible regions. |
+
+    Possible options are:
+
+    | Options | Description |
+    | ------- | ----------- |
+    | `-h` or `--help`| Displays the command help. |
+    | `-i` or `--instance-type` | Google Cloud instance type for the BigAnimal cluster. The help command provides a list of possible VM instance types. Choose the instance type that best suits your application and workload. Choose an instance type in the memory optimized M1, M2, or M3 series for large data sets. Choose from the compute-optimized C2 or C2D series for compute-bound applications. Choose from the accelerator-optimized A2 or A2 ultra for massively parallelized Compute Unified Device Architecture (CUDA) workloads. Choose from the general purpose M5 or M6I series if you don't require memory or compute optimization.|
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/getting_started/preparing_cloud_account/preparing_gcp/index.mdx:62]: we won't support C2D or any accelerator optimized instance types at launch. only the instance types that have a "Yes" in column K here: https://docs.google.com/spreadsheets/d/1WWBAEgJjVWwSIKBoU8z4rIt3x1p8pXMMn7JkahWtYxQ/edit#gid=1228767107
and the general purpose instance families we support at MVP will be E2, N2, and N2D - not m5 or m6i, those are AWS volumes

File: product_docs/docs/biganimal/release/getting_started/preparing_cloud_account/preparing_gcp/understanding_quotas_in_gcp.mdx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: "Understanding requirements in Google Cloud"
+---
+
+Follow these BigAnimal requirements and recommended resource limits in Google Cloud.
+
+
+## vCPU limits
+
+Any time a new VM is deployed in Azure, the vCPUs for the VMs must not exceed the total vCPU limits for the region.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/getting_started/preparing_cloud_account/preparing_gcp/understanding_quotas_in_gcp.mdx:10]: is there a google cloud version of this section yet?

File: product_docs/docs/biganimal/release/overview/03a_region_support/index.mdx [hunk 4]
```diff
@@ -117,3 +177,22 @@ When using AWS and BigAnimal's cloud account, you can create clusters in the fol
 | Europe (Frankfurt) | eu-central-1 |
 | Europe (Ireland)   | eu-west-1    |
 | Europe (London)    | eu-west-2    |
+
+
+### Google Cloud regions
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/overview/03a_region_support/index.mdx:182]: the regions in NA are not correct; see column F here: https://docs.google.com/spreadsheets/d/1FQiRA-OIzX8CWchY1UGxgJvIblUIf5Ie5SSF2FJ_bOE/edit#gid=830197369
currently BAH+GCP supports N Virginia, Belgium, Sydney, Mumbai regions with more coming soon

**Review by amrita-suresh — CHANGES_REQUESTED** [review]

**Pull request merged**